test/sampling.jl

```diff
@@ -265,4 +267,15 @@ test_same(replace=false, ordered=false)
         # This corner case should succeed
         f(view(x, 2:4), view(x, 5:6))
     end
-end
\ No newline at end of file
+end
+
+@testset "issue #872" begin
+    for T in [Int8, Int16, Int32, Int64, Int128, BigInt], f in [identity, unsigned]
+        T == BigInt && f == unsigned && continue
+        T = f(T)
+        samp = sample(T(1):T(10), T(2); replace=false, ordered=false)
```

Comment:
Does
```suggestion
        samp = sample(T(1):T(10), 2; replace=false, ordered=false)
```
also return samples of type `T`? I would have assumed that the type of the second argument does not affect the output type.

Reply:
Indeed. Could also test `sample(1:10, T(2); replace=false, ordered=false)`.

Reply:
Good call, tests for both cases added!